robottelo/host_helpers/api_factory.py

```diff
@@ -618,31 +620,15 @@ def disable_syncplan(self, sync_plan):
         sync_plan = sync_plan.update(['enabled'])
         assert sync_plan.enabled is False
 
-
-class templateupdate:
-    """Context Manager to unlock lock template for updating"""
-
-    def __init__(self, temp):
-        """Context manager that takes entities.ProvisioningTemplate's object
-
-        :param entities.ProvisioningTemplate temp: entities.ProvisioningTemplate's object
-        """
-        self.temp = temp
-        if not isinstance(self.temp, self._satellite.api.ProvisioningTemplate):
-            raise TypeError(
-                'The template should be of type entities.ProvisioningTemplate, {} given'.format(
-                    type(temp)
-                )
-            )
```

Comment:
I intentionally removed it as it was not needed.

Reply:
@jyejare We have had this discussion already. Do not open it again, please.

Reply:
Didnt get that @ogajduse , Can you repeat ? You mean the template types are not just of `ProvisioningTemplate` ?

Also I dont want to block this PR on this comment !

Reply:
@jyejare see this discussion: https://github.com/SatelliteQE/robottelo/pull/10792#discussion_r1122729344

Reply:
@LadislavVasina1 I saw the discussion and @shweta83 comment agreeing on @ogajduse suggestion to not checking the type of template is not quite clearly mentioned there.

May be one of them can tell us ?

Reply:
@jyejare A suggestion not to check type was made because we don't check it elsewhere in our framework. [https://github.com/shweta83/robottelo/blob/edd5d39b16d8262ebe6b4dfd153443760d4a0e6d/robottelo/host_helpers/api_factory.py#L500](https://github.com/SatelliteQE/robottelo/pull/here)

Reply:
@LadislavVasina1 That was not good reason but as I said I dont want to hold this PR on it! This PR is already critical for automation!